autoscheduler/frontend/src/utils/timeUtil.ts

```diff
@@ -36,3 +38,55 @@ export const timeToEvent = (h: number, m: number, offset = 0, clientHeight = 100
     clientX: 100,
   };
 };
+
+interface FirstLastHour {
+  first: number;
+  last: number;
+}
+
+/**
+ * Determines the first and last hour of the schedule, depending on whether
+ * @param schedule The schedule to determine the hours for
+ * @param fullscreen Whether we're fullscreen or not
+ * @returns FirstLastHour enum, containing the first and last hour of the schedule
+ */
+export function getFirstAndLastHour(schedule: Meeting[], fullscreen: boolean): FirstLastHour {
+  let earliest = Number.MAX_VALUE;
+  let latest = Number.MIN_VALUE;
+
+  // Retrieve min/max start & end times
+  schedule.forEach((meeting) => {
+    if (meeting.startTimeHours !== 0 && meeting.startTimeHours < earliest) {
+      earliest = meeting.startTimeHours;
+    }
+
+    if (meeting.endTimeHours !== 0 && meeting.endTimeHours > latest) {
+      latest = meeting.endTimeHours;
+    }
+  });
+
+  const MIN = 8; // The minimum amonut of time between earliest and latest
+
+  // Check to ensure the time between latest and earliest is < MIN, and correct it if not
+  // Also ensure we're not doing this calculation when the schedule is empty
+  if (schedule.length > 0 && latest - earliest < MIN) {
+    // Means our latest class is after 3 - set it to noon
+    if (earliest + MIN >= LAST_HOUR) {
+      // Latest hour always gets an hour added to it, so account for that
+      earliest = latest + 1 - MIN;
+    } else {
+      // Expand it
+      latest = earliest + MIN;
+    }
+  }
+
+  // Use earliest & latest if we're fullscreen
+  let first = FIRST_HOUR;
+  let last = LAST_HOUR;
+  if (fullscreen && schedule.length > 0) {
+    first = earliest;
+    last = latest + 1;
+  }
+
+  return { first, last };
```

Comment:
Since none of the logic above this gets applied unless `fullscreen` is set, I would just return early by checking if it's set at the beginning to make it more clear how these numbers are set